ConnectorCore/BatchUploader.OperationDriver/BatchUploadOperationDriver.cs

```diff
@@ -0,0 +1,129 @@
+using DesktopUI2;
+using DesktopUI2.Models;
+using DesktopUI2.Models.Filters;
+using DesktopUI2.ViewModels;
+using Speckle.BatchUploader.Sdk;
+using Speckle.BatchUploader.Sdk.CommunicationModels;
+using Speckle.BatchUploader.Sdk.Interfaces;
+using Speckle.Core.Api;
+using Speckle.Core.Credentials;
+
+namespace Speckle.BatchUploader.OperationDriver;
+
+/// <summary>
+/// Connector implementation of a BatchUploader client
+/// </summary>
+/// <remarks>
+/// Only processes started by the batch uploader will start executing jobs
+/// </remarks>
+public sealed class BatchUploadOperationDriver
+{
+  private readonly BatchUploaderClient _client;
+  private readonly IApplicationFunctionalityController _applicationController;
+  private readonly ConnectorBindings _connectorBindings;
+  private int _jobCounter;
+
+  public BatchUploadOperationDriver(
+    BatchUploaderClient client,
+    IApplicationFunctionalityController applicationController,
+    ConnectorBindings connectorBindings
+  )
+  {
+    _client = client;
+    _applicationController = applicationController;
+    _connectorBindings = connectorBindings;
+  }
+
+  /// <summary>
+  /// Queries local BatchUploader service for any jobs for this application process
+  /// and starts executing them. Application will terminate once all jobs are complete.
+  /// </summary>
+  public async Task ProcessAllJobs()
+  {
+    while (await GetNextJobId().ConfigureAwait(false) is { } jobId)
+    {
+      // currently using this counter to determine whether this process was started by the batch uploader
+      // and needs to be closed by the batch uploader. Not the most robust system, I'd imagine.
+      _jobCounter++;
+      try
+      {
+        await ProcessJob(jobId).ConfigureAwait(false);
+        await _client.UpdateJobStatus(jobId, JobStatus.Completed).ConfigureAwait(false);
+        //TODO: Finish endpoint with result details.
+        Console.WriteLine("Finished");
+      }
+      catch (Exception ex)
+      {
+        Console.WriteLine(ex);
+        //TODO: Finish endpoint with ex details
+        await _client.UpdateJobStatus(jobId, JobStatus.Failed).ConfigureAwait(false);
+        throw;
+      }
+    }
+
+    if (_jobCounter > 0)
+    {
+      Environment.Exit(0);
```

Comment:
Please check the edge case where batchuploader could be running but no jobs have run.
Would revit just remain open?

Reply:
This is a known "error state" that we plan on handling more explicitly in the Batchuploader desktop app (since it manages the revit subprocess directly)